File: 01193.py
X = int(input())

n = 1
while (n * n + n) // 2 < X:
    n += 1

an = (n * n + n) // 2
dis = an - X
if n % 2:
    up = 1 + dis
    down = n - dis
else:
    up = n - dis
    down = 1 + dis

print(up, '/', down, sep='')

File: 01193.cpp
#include <iostream>
using namespace std;

int main() {
	int X; cin >> X;

	int n = 1;
	while ((n * n + n) / 2 < X) n++;

	int an = (n * n + n) / 2;
	int dis = an - X;
	int up, down;
	if (n % 2) {
		up = 1 + dis;
		down = n - dis;
	}
	else {
		up = n - dis;
		down = 1 + dis;
	}

	cout << up << '/' << down << endl;
}
